DependencyInjection/DoctrineExtension.php

```diff
@@ -438,8 +444,27 @@ protected function loadOrmEntityManager(array $entityManager, ContainerBuilder $
             $this->loadOrmSecondLevelCache($entityManager, $ormConfigDef, $container);
         }
 
-        if ($entityManager['repository_factory']) {
-            $methods['setRepositoryFactory'] = new Reference($entityManager['repository_factory']);
+        if ($entityManager['use_service_repositories']) {
+            if ($entityManager['repository_factory']) {
+                throw new InvalidArgumentException('The "repository_factory" option cannot be set when "use_service_repositories" is set to true.');
+            }
+
+            $serviceRepoFactoryDef = $container->getDefinition('doctrine.orm.container_repository_factory');
+            if (class_exists(ServiceLocatorTagPass::class)) {
+
+                $serviceRepoFactoryDef->replaceArgument(0, new Reference('service_container'));
```

Comment:
This looks wrong. it should be a service locator.

Anyway I think it will be easier to deal with the compat in the compiler pass

Reply:
Good call - I updated to be in the pass. to keep 2.7 support, the class itself accepts a locator or Symfony's `ContainerInterface` for older 3.2 and lower.